### PR TITLE
Call getNearestEventsCardDataCount in getNearestEventsCardData

### LINKED ISSUE #241: [VMS] Call `getNearestEventsCardDataCount` in `getNearestEventsCardData`
* Unlike `getUpcomingEventsCardData()` which first gets called on the server-side and receives the result of `getUpcomingEventsCardDataCount()`, `getNearestEventsCardDataCount()` cannot be called server-side since a user's location is not known.
* This means that `getNearestEventsCardData()` is first called by the client via an API call with some invalid/dummy value for `totalCount` (perhaps -1). You should modify the code in `getNearestEventsCardData` to check if `totalCount` equals this dummy value and call `getUpcomingEventsCardDataCount` if that's the case.
* For clean-cpde purposes, let's also modify `getNearestEventsCardDataCount` arguments to be of type `LocationPageInformation["location"]`. Then you can stop exporting `Coordinates` from `utils/types.ts`

### PR DESCRIPTION
**Pull request by matthewbrown77**
Closes #241

### CODE REVIEW

**Comment by matthewbrown77** [comment]
Output from calling `getNearestEventsCardData` with a count of `-1`:
result

File: server/actions/events.ts [hunk 1]
```diff
@@ -74,6 +73,10 @@ export async function getNearestEventsCardData({
 }: LocationPageInformation): Promise<LocationPaginatedEventCards> {
   await Mongo();
 
+  if (totalCount == -1) {
```
Comment (PC98) [inline server/actions/events.ts:76]: for future reference, consider using strict equality `===` 🙂

**Review by PC98 — APPROVED** [review]

File: server/actions/events.ts (hunk 1 above)
Comment (matthewbrown77) [reply server/actions/events.ts:76]: Is using strict equality necessary with typescript?

Comment (PC98) [reply server/actions/events.ts:76]: Not really, but I guess this is a general rule of thumb when working with a mixed codebase (JS and TS) and a language that supports both strict and non-strict equality operators.

**Pull request merged**